### DISCUSSION TITLE
portaudio / coreaudio device specification by name and not port?

### DISCUSSION THREAD

**Discussion by sogorman** [CESNET/UltraGrid#385] [Q&A] [answered]
In OSX, port audiodevice IDs change form time to time. Currently before stating UG I have to have a script interrogate the current port audio indexes to get the correct index ID for the device I want to use. Would it be possible and or beneficial to specify the device name?

 ```
(base) sogorman@Seans-MacBook-Pro-4 MacOS % ./uv -s coreaudio:help 
UltraGrid 1.8.6 (tags/v1.8.6 rev 6328048 built Oct 20 2023 14:32:46)

Available Core Audio capture devices:
	coreaudio    : Default CoreAudio capture
	coreaudio:97 : Sean?s iPhone 14 Microphone
	coreaudio:92 : MacBook Pro Microphone
	coreaudio:75 : Microsoft Teams Audio
Exit
(base) sogorman@Seans-MacBook-Pro-4 MacOS % 

```

so if I wnated to use my macbook pro microphone I would use 

```
(base) sogorman@Seans-MacBook-Pro-4 MacOS % ./uv -s coreaudio:92  
```

It would be awesome to call UG like this

```
(base) sogorman@Seans-MacBook-Pro-4 MacOS % ./uv -s coreaudio:"MacBook Pro Microphone"
```

**Comment by MartinPulec** [comment, marked as answer]
Actually, it is already implemented in [1.9.x](https://github.com/CESNET/UltraGrid/releases/tag/v1.9.2) versions (obviously also in continuous):
```
UltraGrid 1.9.2 (tags/v1.9.2 rev 0d8eaf2 built Apr 29 2024 11:16:31)

Core Audio capture usage:
        -s coreaudio[:<index>|:<name>]
where
        <name> - device name substring (case sensitive)

Available Core Audio capture devices:
        coreaudio    : Default CoreAudio capture
        coreaudio:76 : BlackHole 16ch
        coreaudio:91 : MacBook Pro Microphone
Exit
```

The usage is exactly as you write – `-s coreaudio:"MacBook Pro Microphone"`. (Note that I've noticed that the name doesn't work for the default one for now. I'll fix it, but it shouldn't be a big deal since you can still use `-s coreaudio` for the default.)
